app/scripts/directives/createSecret.js

```diff
@@ -143,6 +143,17 @@ angular.module("openshiftConsole")
         };
 
         var linkSecretToServiceAccount = function(secret) {
+          if (!$scope.serviceAccountsNames.contains($scope.newSecret.pickedServiceAccountToLink)) {
+            var alert = {
+              name: 'createAndLink',
+              data: {
+                type: "success",
+                message: "Secret " + secret.metadata.name + " was created but couldn't be linked with service account " + $scope.newSecret.pickedServiceAccountToLink + "."
```

Comment:
can we not break this into two steps (Create, Link) and then do a better job of reporting the result of each aspect?